assets/app/scripts/directives/logViewer.js

```diff
@@ -269,6 +269,11 @@ angular.module('openshiftConsole')
                 streamer = null;
                 $scope.$evalAsync(function() {
                   $scope.autoScrollActive = false;
+                  // if no logs, they have already been archived.
+                  if(lastLineNumber === 0) {
+                    $scope.state = 'empty';
+                    $scope.emptyStateMessage = 'The logs are no longer available or could not be loaded.';
```

Comment:
1.)  Cancelling a `Build` then clicking logs will get a WS 400 error.  The streamer `.onError` will fire, followed by the `.onClose`.  
![build_cancel_ws_400](https://cloud.githubusercontent.com/assets/280512/13153819/dba2cccc-d643-11e5-8153-e0b752616c07.png)

2.) Cancelling a `Deployment` then clicking the `logs` tab does not generate a 400 error.  In this case, only the streamer `.onClose` event will fire.
![deployment_cancel_ws_closed](https://cloud.githubusercontent.com/assets/280512/13153893/4394f1f2-d644-11e5-8bbb-a8f8015173ff.png)

3.)  Failed `Pod` only triggers `.onClose`.
![pod_failed_closed](https://cloud.githubusercontent.com/assets/280512/13153931/75ba2eae-d644-11e5-8ccf-9b9a0f9e3729.png)

So far in these 3 cases only the build causes a socket error.